### PR TITLE
Add git SHA to binaries at build time

### PR DESCRIPTION
**Pull request by viveksyngh**
Signed-off-by: Vivek Singh <vivekkmr45@yahoo.in>

Add git SHA to binaries at build time

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/alexellis/derek/blob/master/CONTRIBUTING.md))
#10 

## How Has This Been Tested?

Tested on my local. 
```
make                  
GOOS=linux go build -o license-check --ldflags "-s -w \
	-X main.GitCommit=7c87f1b83dc8d3488c090afb76bd0c03e787a388" \
	-a -installsuffix cgo
GOOS=darwin go build -o license-check-darwin --ldflags "-s -w \
	-X main.GitCommit=7c87f1b83dc8d3488c090afb76bd0c03e787a388" \
	-a -installsuffix cgo
GOOS=linux GOARCH=arm GOARM=6 go build -o license-check-armhf --ldflags "-s -w \
	-X main.GitCommit=7c87f1b83dc8d3488c090afb76bd0c03e787a388" \
	-a -installsuffix cgo
GOOS=linux GOARCH=arm64 go build -o license-check-arm64 --ldflags "-s -w \
	-X main.GitCommit=7c87f1b83dc8d3488c090afb76bd0c03e787a388" \
	-a -installsuffix cgo
GOOS=linux GOARCH=s390x go build -o license-check-s390x --ldflags "-s -w \
	-X main.GitCommit=7c87f1b83dc8d3488c090afb76bd0c03e787a388" \
	-a -installsuffix cgo
GOOS=linux GOARCH=ppc64le go build -o license-check-ppc64le --ldflags "-s -w \
	-X main.GitCommit=7c87f1b83dc8d3488c090afb76bd0c03e787a388" \
	-a -installsuffix cgo


./license-check-darwin -version 
Commit: 18c0cc6c9fdcb4398bc40dbcadca5f8302f0975c
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/alexellis/derek/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -12,6 +12,8 @@ import (
 
 var (
 	verbose bool
+	//GitCommit git commit SHA
+	GitCommit string
 )
 
 func main() {
```
Comment (alexellis) [inline main.go:19]: Hi @viveksyngh 

Thank you for this PR!

Do we need to use this `GitCommit` in the version command or in the `main()`?

What I meant was to add SHASUM of the binary when run CI/CD and generate a release, but I think this is a great addition too and I would also like to merge this 👍 

Alex

**Review by alexellis — CHANGES_REQUESTED** [review]
Sending feedback

File: main.go (hunk 1 above)
Comment (viveksyngh) [reply main.go:19]: We can print this in main while running in verbose mode.

**Comment by alexellis** [comment]
How about some `-version` flag that prints, then exits?

**Comment by viveksyngh** [comment]
I think that is better.

**Comment by viveksyngh** [comment]
I have added script to generate SHASUM.

**Comment by viveksyngh** [comment]
I have also added `-version` flag as well.

**Comment by alexellis** [comment]
Perhaps make should invoke the hashgen?

**Comment by alexellis** [comment]
@viveksyngh thank you for this PR.

I think as a follow-up we should complete #4 and make sure that we invoke the hashgen script at that time. This will mean the binaries will be automatically uploaded whenever I do a release on the repo. 

Alex

**Pull request merged**